ganga/GangaCore/Lib/Virtualization/Docker.py

```diff
@@ -0,0 +1,26 @@
+##########################################################################
+# Ganga Project. http://cern.ch/ganga
+#
+##########################################################################
+from GangaCore.GPIDev.Schema import Schema, Version, SimpleItem
+from GangaCore.GPIDev.Adapters.IVirtualization import IVirtualization
+import GangaCore.Utility.logging
+
+logger = GangaCore.Utility.logging.getLogger()
+
+class Docker(IVirtualization):
+
+    """
+    Handles all the config related to Docker
+    """
+
+    _name = 'Docker'
+    _schema = Schema(Version(1, 0), {
+        'imageUrl': SimpleItem(defvalue="", doc='Link to the container image'),
+        'mode': SimpleItem(defvalue="P1", doc='Mode of container execution'
```

Comment:
This mode attribute only exists in the Docker class however `LocalHost` has to have some knowledge about if this attribute is present or not with a conditional at the following line:
https://github.com/ganga-devs/ganga/pull/1490/files#diff-39811eca09c833c4f431c0bddee31ca9R241

This sort of two way knowledge is a bit clumsy as requires ever greater conditional checking, doesn't scale well and makes debugging harder. I would suggest having an abstract method defined in the `IVirtualization` class and implemented in the concrete derived classes that can add whatever is needed to the script. That way the `LocalHost` backend doesn't care what the virtualisation is, it merely requests any script additions.

Reply:
created getters in IVirtualization class and I overide them in Docker and SIngularity class